src/lib/stitching/exchange/link.ts

```diff
@@ -35,21 +35,19 @@ export const createExchangeLink = () => {
 
   const analyticsMiddleware = setContext((_request, context) => {
     const locals = context.graphqlContext && context.graphqlContext.res.locals
-    if (!locals) return context.graphqlContext
-    return {
-      ...context.graphqlContext,
-      headers: {
-        ...context.graphqlContext.headers,
-        "User-Agent": locals.userAgent
-          ? locals.userAgent + "; Metaphysics"
-          : "Metaphysics",
-      },
+    if (!locals) return context
+    const headers = {
+      ...context.headers,
+      "User-Agent": locals.userAgent
+        ? locals.userAgent + "; Metaphysics"
+        : "Metaphysics",
     }
+    return { headers }
```

Comment:
Should this not include the rest of the context?

```suggestion
    return { ...context, headers }
```

Reply:
Ah ok. I find that a little unintuitive, but definitely no blocker 👍

Reply:
Would this make sure that we also send requestIDs also? I've not seen a metaphysics -> jaws connection in the dd service map and wondered if it was related

Reply:
Just double-checked. Yep, the request ids are going through. I added an extra middleware to make sure.

![image](https://user-images.githubusercontent.com/1242537/48790699-e053f800-ece7-11e8-864b-5d0d486507a8.png)